metaflow/client/core.py

```diff
@@ -878,6 +879,71 @@ def tarball(self) -> tarfile.TarFile:
         """
         return self._tar
 
+    @property
+    def extract(self) -> TemporaryDirectory:
+        """
+        Extracts the code package to a temporary directory.
+
+        This creates a temporary directory containing all user code
+        files from the code package. The temporary directory is
+        automatically deleted when the returned TemporaryDirectory
+        object is garbage collected or when its cleanup() is called.
+
+        To preserve the contents to a permanent location, use
+        os.replace() which performs a zero-copy move on the same
+        filesystem:
+
+        ```python
+        with task.code.extract as tmp_dir:
+            # Move contents to permanent location
+            for item in os.listdir(tmp_dir):
+                src = os.path.join(tmp_dir, item)
+                dst = os.path.join('/path/to/permanent/dir', item)
+                os.makedirs(os.path.dirname(dst), exist_ok=True)
+                os.replace(src, dst)  # Atomic move operation
+        ```
+        Returns
+        -------
+        TemporaryDirectory
+            A temporary directory containing the extracted code files.
+            The directory and its contents are automatically deleted when
+            this object is garbage collected.
+        """
+        exclusions = [
+            "metaflow/",
```

Comment:
I think we have constants for this so we should use that. +1 to factoring it out as well.

Reply:
why would we need to factor this out? #2238 can directly invoke this method unless i am missing something else?

Reply:
Recommend waiting on #2269 which will make this easier.

Reply:
So, discussion was:
- release now
- later will be made easier by 2269
- we can add a comment about where condav2-1.cnd comes from (but yes, it should be included)